### PR TITLE
Colors as data

### PR DESCRIPTION
**Pull request by pontaoski**
This introduces colors as a primitive type in Nushell.

Currently lacking anything to use it because I still need to figure that part out.

### CODE REVIEW

File: crates/nu-protocol/src/value/primitive/color.rs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+use serde::{Deserialize, Serialize};
+use std::hash::{Hash};
+use std::str::FromStr;
+use bigdecimal::BigDecimal;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Deserialize, Serialize)]
+pub enum ColorData {
+    Rgba(Rgb, u8),
+    Hsla(Hsl, u8),
+    Hsva(Hsv, u8)
```
Comment (jonathandturner) [inline crates/nu-protocol/src/value/primitive/color.rs:11]: Can we convert from one color format to another? If so, we might want to just pick one format and store in that one, and then offer color conversion commands to get to the other ones.

**Comment by jonathandturner** [comment]
@pontaoski - are you still interested in working on this?

**Comment by pontaoski** [comment]
> @pontaoski - are you still interested in working on this?

Yes; I just haven't found the time to work on this

File: crates/nu-protocol/src/value/primitive/color.rs (hunk 1 above)
Comment (pontaoski) [reply crates/nu-protocol/src/value/primitive/color.rs:11]: We could, but not all colour space conversions are lossless

**Review by Josegto1 — APPROVED** [review]

**Comment by jonathandturner** [comment]
Going ahead and closing as this is now out of date with the current codebase.

**Pull request closed without merging**